velox/functions/lib/window/NthValue.cpp

```diff
@@ -82,12 +87,28 @@ class NthValueFunction : public exec::WindowFunction {
     auto rawFrameStarts = frameStarts->as<vector_size_t>();
     auto rawFrameEnds = frameEnds->as<vector_size_t>();
 
+    vector_size_t minFrame = rawFrameStarts[0];
+    if (ignoreNulls_) {
+      if (validRows.hasSelections()) {
+        // IgnoreNulls really only needs nullsBuffer information of the vector.
```

Comment:
Can we implement this optimization now? The vector may be expensive to extract especially if it is of complex type.

Reply:
Yes, will do this. Have a PR. Will send it out post writing tests.